.github/workflows/ci.yaml

```diff
@@ -67,6 +80,21 @@ jobs:
       #     key: upstream_ws-${{ env.CACHE_PREFIX }}-${{ hashFiles('moveit2.repos') }}-${{ github.run_id }}
       #     restore-keys: |
       #       upstream_ws-${{ env.CACHE_PREFIX }}-${{ hashFiles('moveit2.repos') }}
+=======
+      - name: Get latest release date for rosdistro
+        id: rosdistro_release_date
+        uses: JafarAbdi/latest-rosdistro-release-date-action@main
+        with:
+          rosdistro: ${{ matrix.env.ROS_DISTRO }}
+      - name: Cache upstream workspace
+        uses: pat-s/always-upload-cache@v2.1.5
+        with:
+          path: ${{ env.BASEDIR }}/upstream_ws
+          key: ${{ env.CACHE_PREFIX }}-${{ github.run_id }}
+          restore-keys: ${{ env.CACHE_PREFIX }}
+        env:
+          CACHE_PREFIX: ${{ steps.rosdistro_release_date.outputs.date }}-upstream_ws-${{ matrix.env.IMAGE }}-${{ hashFiles('moveit2*.repos', '.github/workflows/ci.yaml') }}
```

Comment:
@JafarAbdi what do you think is the best way to resolve this conflict? Should we re-enable upstream_ws caches or simply set the env variable from inside the `rosdistro_release_date` job and append the `.repos` hash only if needed?

Reply:
I re-enabled upstream_ws caches hopefully it will help with speeding up the CI time